pkg/table/tables/partition.go

```diff
@@ -1149,11 +1149,77 @@ func (lp *ForListPruning) locateListColumnsPartitionByRow(tc types.Context, ec e
 		if lp.defaultPartitionIdx >= 0 {
 			return lp.defaultPartitionIdx, nil
 		}
-		return -1, table.ErrNoPartitionForGivenValue.GenWithStackByArgs("from column_list")
+		indices := make([]int, 0, len(lp.ColPrunes))
+		for _, colPrune := range lp.ColPrunes {
+			indices = append(indices, colPrune.ExprCol.Index)
+		}
+		valueMsg, err := formatPartitionValues(r, indices)
+		if err != nil {
+			return -1, errors.Trace(err)
+		}
+		return -1, table.ErrNoPartitionForGivenValue.GenWithStackByArgs(valueMsg)
 	}
 	return location[0].PartIdx, nil
 }
 
+func formatPartitionValues(row []types.Datum, indices []int) (string, error) {
+	if len(indices) == 0 {
+		return "from column_list", nil
+	}
+	values := make([]string, 0, len(indices))
+	for _, idx := range indices {
+		if idx < 0 || idx >= len(row) {
+			return "", errors.Errorf("partition column index %d out of range", idx)
+		}
+		formatted, err := formatPartitionDatum(row[idx])
```

Comment:
Could `genIndexKeyStrs()` be used instead? Similar to how Duplicated Key Error is generated in `buildIndexForRow()`.